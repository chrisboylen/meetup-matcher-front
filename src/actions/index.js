export const displayMeetups = (meetups) => ({
  type: 'DISPLAY_MEETUPS',
  meetups
});

export const loginUser = (user) => ({
  type: 'LOGIN_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const userError = (message) => ({
  type: 'USER_ERROR',
  message
});