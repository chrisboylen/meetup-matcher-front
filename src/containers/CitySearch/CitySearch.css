h1, h3 {
  color: #fef5f5;
  text-align: center; }

.city-search {
  background-color: rgba(80, 66, 65, 0.8);
  border-radius: 10px;
  margin: 2rem;
  padding: 2rem; }

form {
  display: flex;
  justify-content: center;
  flex-wrap: wrap; }
  form input, form button {
    font-size: 1.2rem;
    margin: 0 0 1rem 0;
    min-height: 3rem;
    padding: 0 1.5rem 0 1rem; }
  form input {
    background-color: rgba(12, 8, 10, 0.7);
    border: 0;
    color: #fef5f5; }
  form .city-input {
    border-radius: 10px 0 0 10px;
    margin: 0 .3rem 1rem 0; }
  form button {
    background-color: #0c090a;
    border: 2px #0c090a;
    border-radius: 0 10px 10px 0;
    color: #ff2600;
    height: 3rem; }
