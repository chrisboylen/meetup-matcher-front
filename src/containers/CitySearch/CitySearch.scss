$mainWhite: #fef5f5;

h1, h3 {
  color: $mainWhite;
  margin: 0;
  text-align: center;
}

.city-search {
  background-color: #5c5252cc;
  border-radius: 10px;
  margin: 2rem;
  padding: 2rem;
}

form {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  margin: 1rem 0 0 0;
    input, button {
      font-size: 1.2rem;
      margin: 0 0 1rem 0;
      min-height: 3rem;
      padding: 0 1.5rem 0 1rem;
    }
    input {
      background-color: rgba(12, 8, 10, 0.7);
      border: 0;
      color: $mainWhite;
    }
    .city-input {
      border-radius: 10px 0 0 10px;
      margin: 0 .3rem 1rem 0;
    }
    button {
      background-color: #0c090a;
      border: 2px #0c090a;
      border-radius: 0 10px 10px 0;
      color: rgba(255, 38, 0, 1);
      height: 3rem;
    }
}