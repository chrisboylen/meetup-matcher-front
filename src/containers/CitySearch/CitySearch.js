import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux'; 
import { PropTypes } from 'prop-types';
import MeetupsCont from '../MeetupsCont/MeetupsCont';
import { fetchPopularMeetups } from '../../Utilities/fetchData';
import { displayMeetups } from '../../actions';
import './CitySearch.css'

export class CitySearch extends Component {
  constructor () {
    super();
    this.state = {
      city: '',
      state: ''
    };
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({ [name]: value });
  }

  handleSubmit = async (e) => {
    const { displayMeetups } = this.props;
    e.preventDefault();
    const { city, state } = this.state;
    const meetups = await fetchPopularMeetups(city, state);

    displayMeetups(meetups);
    this.setState({
      city: '',
      state: ''
    })
  }

  render() {
    const { city, state } = this.state;
    const { meetups } = this.props;

    return (
      <div className="city-search">
        <h1>Welcome to Meetup Matcher</h1>
        <h3>Enter your destination to find meetups.</h3>
        <form onSubmit={this.handleSubmit}>
          <input 
            className="search city-input"
            type="text"
            name="city"
            value={city}
            placeholder="Enter city"
            onChange={this.handleChange}
          />
          <input 
            className="search state-input"
            type="text"
            name="state"
            value={state}
            placeholder="Enter state"
            onChange={this.handleChange}
          />
          <button>SEARCH</button>
        </form>
        {meetups && <MeetupsCont />}
      </div>
    );
  }
}

CitySearch.propTypes = {
  displayMeetups: PropTypes.func,
  meetups: PropTypes.array
};

export const mapStateToProps = (state) => ({
  meetups: state.meetups
});

export const mapDispatchToProps = (dispatch) => ({
  displayMeetups: (meetups) => dispatch(displayMeetups(meetups))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(CitySearch));

