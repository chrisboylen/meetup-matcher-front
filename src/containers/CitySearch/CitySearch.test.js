import React from 'react';
import { shallow } from 'enzyme';
import { CitySearch } from './CitySearch';
import { displayMeetups } from '../../actions';

describe('CitySearch', () => {
  let wrapper;
  let mockEvent;

  beforeEach(() => {
    mockEvent = {target: {value: 'bingo'}};
    wrapper = shallow(<CitySearch />);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<CitySearch />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should set state when invoked', () => {
      mockEvent = {target: {name: 'city', value: 'bingo'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().city).toBe('bingo');

      mockEvent = {target: {name: 'state', value: 'bongo'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().state).toBe('bongo');
    });
    it('should invoke handleChange when city is changed', () => {
      const spy = spyOn(wrapper.instance(), 'handleChange');
      wrapper.instance().forceUpdate();
      wrapper.find('.city-input').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalled();
    });
    it('should invoke handleChange when state is changed', () => {
      const spy = spyOn(wrapper.instance(), 'handleChange');
      wrapper.instance().forceUpdate();
      wrapper.find('.state-input').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalled();
    });
  });

  describe('handleSubmit', () => {
    let mockDisplayMeetups;

    beforeEach(() => {
      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({json: () => Promise.resolve({name: 'bingo'})}));

      mockDisplayMeetups = jest.fn();
      mockEvent = {preventDefault: jest.fn()};
      wrapper = shallow(
        <CitySearch displayMeetups={mockDisplayMeetups} />
      );
    });
    it('should invoke handleSubmit when form is submitted', () => {
      const spy = spyOn(wrapper.instance(), 'handleSubmit');
      wrapper.instance().forceUpdate();
      wrapper.find('button').simulate('click', mockEvent);

      expect(spy).toHaveBeenCalled();
    });
  });
});