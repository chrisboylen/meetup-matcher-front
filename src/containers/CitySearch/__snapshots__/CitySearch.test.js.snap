// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitySearch should match snapshot 1`] = `
<div
  className="city-search"
>
  <h1>
    Welcome to Meetup Matcher
  </h1>
  <h3>
    Enter your destination to find meetups.
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      className="search city-input"
      name="city"
      onChange={[Function]}
      placeholder="Enter city"
      type="text"
      value=""
    />
    <input
      className="search state-input"
      name="state"
      onChange={[Function]}
      placeholder="Enter state"
      type="text"
      value=""
    />
    <button>
      SEARCH
    </button>
  </form>
</div>
`;
